frontend/src/app/login/login.component.html

```diff
@@ -0,0 +1,39 @@
+<div>
+    <form [formGroup]="loginForm" (submit)="login(loginForm.value)">
+        <h4>Login</h4>
+        <div class="form-group">
+            <label>Username</label>
+            <ng-container *ngIf="displayUsernameError && username.invalid && (username.dirty || username.touched)">
```

Comment:
these long boolean statements could be moved to methods on the component. Not absolutely necessary, just a suggestion